### ISSUE TITLE
write new line at the of a JSON lines stream

### ISSUE THREAD

**Issue by zazi** [xbib/marc#3]
Currently, when I attach multiple files of JSON lines output, I get the last record of the previous file and the first record of the current one (which breaks the JSON lines format, because then two records are in one line).
I guess, it's simply adding a new line in MarcJsonWriter at https://github.com/xbib/marc/blob/master/src/main/java/org/xbib/marc/json/MarcJsonWriter.java#L299, but I'm not sure + I'm not that involved in your release management (hence, this issue here ;) ).

**Comment by jprante** [comment]
A line feed is written only in Style.ELASTICSEARCH_BULK after each record. This style is meant for writing JSON lines no matter how many input files were given.

You can also use the Style.ARRAY to join MARC files into a single JSON.  Note the use of `writer.beginCollection()`, `writer.endCollection()`, and the `writeRecords()` method in the `Marc.builder()` instance.

```
        FileOutputStream out = new FileOutputStream(file);
        try (MarcJsonWriter writer = new MarcJsonWriter(out, MarcJsonWriter.Style.ARRAY)) {
            writer.beginCollection();
            try (InputStream inputStream = getClass().getResource("/org/xbib/marc/summerland.mrc").openStream()) {
                Marc.builder()
                        .setFormat(MarcXchangeConstants.MARCXCHANGE_FORMAT)
                        .setType(MarcXchangeConstants.BIBLIOGRAPHIC_TYPE)
                        .setInputStream(inputStream)
                        .setCharset(Charset.forName("ANSEL"))
                        .setMarcRecordListener(writer)
                        .build()
                        .writeRecords();
            }
            try (InputStream inputStream = getClass().getResource("/org/xbib/marc/chabon.mrc").openStream()) {
                Marc.builder()
                        .setFormat(MarcXchangeConstants.MARCXCHANGE_FORMAT)
                        .setType(MarcXchangeConstants.BIBLIOGRAPHIC_TYPE)
                        .setInputStream(inputStream)
                        .setCharset(Charset.forName("ANSEL"))
                        .setMarcRecordListener(writer)
                        .build()
                        .writeRecords();
            }
            writer.endCollection();
        }
```

This is still on one line but valid JSON.

The formats Style.LINES, which is the default, and Style.ARRAY do not support line feed after a record.
To ease the readbility, I added a method `jsonWriter.writeLine()` that adds an extra line feed in all JSON writer styles.

For example, to add a line feed between input files, you can use 

```
        FileOutputStream out = new FileOutputStream(file);
        try (MarcJsonWriter writer = new MarcJsonWriter(out, MarcJsonWriter.Style.ARRAY)) {
            writer.beginCollection();
            try (InputStream inputStream = getClass().getResource("/org/xbib/marc/summerland.mrc").openStream()) {
                Marc.builder()
                        .setFormat(MarcXchangeConstants.MARCXCHANGE_FORMAT)
                        .setType(MarcXchangeConstants.BIBLIOGRAPHIC_TYPE)
                        .setInputStream(inputStream)
                        .setCharset(Charset.forName("ANSEL"))
                        .setMarcRecordListener(writer)
                        .build()
                        .writeRecords();
            }
            writer.writeLine();
            try (InputStream inputStream = getClass().getResource("/org/xbib/marc/chabon.mrc").openStream()) {
                Marc.builder()
                        .setFormat(MarcXchangeConstants.MARCXCHANGE_FORMAT)
                        .setType(MarcXchangeConstants.BIBLIOGRAPHIC_TYPE)
                        .setInputStream(inputStream)
                        .setCharset(Charset.forName("ANSEL"))
                        .setMarcRecordListener(writer)
                        .build()
                        .writeRecords();
            }
            writer.endCollection();
        }
```

I have released Version 1.2.1 with that enhancement.

**Comment by zazi** [comment]
yeah, it's working, see https://github.com/slub/marc2jsonl/pull/7 ;)

thx a lot @jprante for this enhancement.
